### PR TITLE
Remove Thanos ZLabel type in favour of identical LabelAdapter type 

### PR DESCRIPTION
**Pull request by bboreham**
This code was brought in with #3222, but exactly duplicated code we already had, except for the name.

(`LabelAdapter` isn't a hugely descriptive name, but it was widely used already)

All the types in `storegateway/labelpb` were removed: `Label`, `LabelSet` and `ZLabelSet`; the last two were entirely unused in Mimir.  
`Label` was equivalent to `LabelPair` in Mimir, albeit using `string` instead of `bytes` type.  According to [here](https://groups.google.com/g/protobuf/c/au6eQBkRT5s) these are identical on the wire, but `string` should only contain valid UTF-8 data.  (This is not checked in Go).

#### Checklist

- [x] Tests updated
- NA Documentation added
- NA `CHANGELOG.md` updated - not user-facing

### CODE REVIEW

File: pkg/storegateway/storepb/types.proto [hunk 1]
```diff
@@ -31,7 +31,7 @@ message Chunk {
 }
 
 message Series {
-  repeated Label labels      = 1 [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/grafana/mimir/pkg/storegateway/labelpb.ZLabel"];
+  repeated cortexpb.LabelPair labels      = 1 [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/grafana/mimir/pkg/mimirpb.LabelAdapter"];
```
Comment (colega) [inline pkg/storegateway/storepb/types.proto:34]: Why do we need the change here to `cortexpb.LabelPair`?

Comment (bboreham) [reply pkg/storegateway/storepb/types.proto:34]: Because I deleted Label.

Comment (colega) [reply pkg/storegateway/storepb/types.proto:34]: Oh it was in types.proto. Sorry I missed that one.

Comment (colega) [reply pkg/storegateway/storepb/types.proto:34]: So the proto will unmarshal even if the type name is different? In that case, should we create a task to rename `cortexpb` to `mimirpb`?

Comment (bboreham) [reply pkg/storegateway/storepb/types.proto:34]: I added more info to the PR description.

**Comment by pracucci** [comment]
@bboreham I've just merged #3288. Could you rebase, please?

**Review by pracucci — APPROVED** [review]
LGTM!

**Pull request merged**